application/engine/services/context_budget_allocator.py

```
@@ -1230,41 +1260,78 @@ def _get_vector_recall(
+            # 旧的 collection 名称（带重复 novel- 前缀）
+            legacy_collection = f"novel_{novel_id}_chunks"
+
+            # 获取已有 collection 列表用于回退判断
+            existing_collections = []
+            try:
+                import asyncio
+                async def _list():
+                    return await self.vector_facade.vector_store.list_collections()
+                existing_collections = asyncio.run(_list())
+            except Exception:
+                pass
```

Ruff (0.15.10)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)